helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -393,6 +394,12 @@ public void close() {
     return finalIdealStateMap;
   }
 
+  protected List<HelixRebalanceException.Type> failureTypesToPropagate() {
+    // By default, only propagate the following types of HelixRebalanceException
+    return Arrays.asList(HelixRebalanceException.Type.INVALID_REBALANCER_STATUS,
```

Comment:
nit, let's define a static unmodifiable list for the WagedRebalancer class to avoid creating a list on each call. Then we can always create a new list in the ReadOnlyWagedRebalancer.java. Considering ReadOnlyWagedRebalancer.java is less used and does not impact the critical pipeline time, it is better to create new list there.